Remove feature "Case Detail Print" along with it's feature flag

mkangia opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->
Removes the feature for "Case detail print" originally added in https://github.com/dimagi/commcare-hq/pull/15539

## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->
https://dimagi.atlassian.net/browse/SAAS-18095

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->
`CASE_DETAIL_PRINT`

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
Removed code in clear steps, will be doing sanity testing locally.
Tested locally for
1. previous apps using and have set print template for case detail working okay after code changes.

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
QA requested for sanity testing of app builder
https://dimagi.atlassian.net/browse/QA-7962

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

mkangia committed:
remove views for case detail print feature

mkangia committed:
remove UI for case detail print

mkangia committed:
remove print template from detail app and xml model

mkangia committed:
remove print_template in case list overwrite feature

mkangia committed:
remove setting of print_id and hence print_id

mkangia committed:
remove related usage of print_template

mkangia committed:
remove tests for case detail print

mkangia committed:
remove print_template from test & static files

mkangia committed:
remove saving print_template

mkangia committed:
remove feature flag for case detail print

mkangia committed:
Merge branch 'master' into mk/18095-remove-case-detail-print

mkangia committed:
Merge branch 'master' into mk/18095-remove-case-detail-print

mkangia commented on corehq/apps/app_manager/suite_xml/xml_models.py:
```diff
@@ -865,7 +865,6 @@ class Field(OrderedXmlObject):
     ORDER = ('style', 'header', 'template', 'endpoint_action', 'sort_node', 'alt_text')
 
     sort = StringField('@sort')
-    print_id = StringField('@print-id')
```
Not sure I understand.
We are not passing the print_id anymore.
https://github.com/dimagi/commcare-hq/pull/36764/files#diff-82f8a21538bd233de16765fa32282aeba9635783958669bd5643a480b37d2ec2

Does that answer your question?

kaapstorm reviewed (approved):
Thanks for clarifying.

orangejenny commented on corehq/apps/app_manager/views/modules.py:
```diff
@@ -114,12 +114,6 @@
 from corehq.apps.domain.models import Domain
 from corehq.apps.fixtures.fixturegenerators import item_lists_by_app, REPORT_FIXTURE, LOOKUP_TABLE_FIXTURE
 from corehq.apps.fixtures.models import LookupTable
-from corehq.apps.hqmedia.controller import MultimediaHTMLUploadController
```
I think you can delete the `MultimediaHTMLUploadController` altogether, including `hqmedia/uploader/preview_html_single.html`.

mkangia replied:
Great! Happy to do so,

https://github.com/dimagi/commcare-hq/pull/36764/commits/a9b71fcf58376364a3110508ab9c213ed6ab5d6e

orangejenny commented on corehq/apps/app_manager/xform.py:
```diff
@@ -729,8 +729,7 @@ def odk_intents(self):
         # but current text references aren't language-specific
-        nodes = self.findall('{h}head/{odk}intent[@class="org.commcare.dalvik.action.PRINT"]'
-                             '/{f}extra[@key="cc:print_template_reference"]')
+        nodes = self.findall('{h}head/{odk}intent[@class="org.commcare.dalvik.action.PRINT"]')
```
I think you can instead delete this function, and the calling code [here](https://github.com/dimagi/commcare-hq/blob/bbb06e7b0c5ed94064e8e6094a7467fe3af4e656/corehq/apps/hqmedia/models.py#L746-L748), which looks like it's for including print templates as part of the app's multimedia.

mkangia replied:
Thanks for the suggestion here @orangejenny 

Before going ahead with the removal I am trying to get some confidence.

I don't have context around this code here, so I am looking at history of updates to this function.

I see an [update](https://github.com/dimagi/commcare-hq/pull/7764/files) to it, way before case detail print feature was added, so I am assuming that this is probably used elsewhere too?
I am thinking of the [form print feature](https://github.com/dimagi/commcare-hq/pull/7594) which was added roughly at the same time as the update I linked above, which could be using this function here.

mkangia replied:
oh wow, well, I assumed "print_template_reference" is for "print_template".
But there is enough proof that I should put it back.

mkangia committed:
remove redundant controller

mkangia committed:
Revert "remove related usage of print_template"

This reverts commit 232620f950d3300eaddbd20d53198c347d7658c1.

print_template_reference was found to be added earlier in
https://github.com/dimagi/Vellum/pull/502 and should still be in use for
printing forms

orangejenny reviewed (commented):
Not approving because I didn't review closely enough, but noting that everything I commented on is addressed.

mkangia committed:
add back longScreen

accidentally removed in 02faf701727e9cd96d27600b8fa3124431500bbb when removing UI for case detail print

kaapstorm reviewed (commented):
Jenny's review taught me that I don't really know enough about this feature, and app multimedia, to confidently approve this. But the change looks good to me.

millerdev reviewed (approved):
Looks reasonable, but I also don't have a lot of context in this area. Trusting that your local testing and QA is thorough.

mkangia commented:
Thanks @millerdev & @kaapstorm 

QA is through and all looks set to go.
